From bd662c090bd4a45cc830de9e42e96dd0f8e1f702 Mon Sep 17 00:00:00 2001
From: NIIBE Yutaka <gniibe@fsij.org>
Date: Thu, 6 May 2021 12:35:19 +0900
Subject: [PATCH] ecc: Fix the previous commit.

* cipher/ecc-misc.c (_gcry_ecc_mont_decodepoint): Fix the condition.

--

GnuPG-bug-id: 5423
Signed-off-by: NIIBE Yutaka <gniibe@fsij.org>
---
 cipher/ecc-misc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cipher/ecc-misc.c b/cipher/ecc-misc.c
index b89dcfa6..0c387c27 100644
--- a/cipher/ecc-misc.c
+++ b/cipher/ecc-misc.c
@@ -331,7 +331,7 @@ _gcry_ecc_mont_decodepoint (gcry_mpi_t pk, mpi_ec_t ctx, mpi_point_t result)
       rawmpi = _gcry_mpi_get_buffer (pk, nbytes, &rawmpilen, NULL);
       if (!rawmpi)
         return gpg_err_code_from_syserror ();
-      if (rawmpilen > nbytes + 1)
+      if (rawmpilen > nbytes + BYTES_PER_MPI_LIMB)
         {
           xfree (rawmpi);
           return GPG_ERR_INV_OBJ;
-- 
2.30.2

